pkg/kubectl/scale.go

```diff
@@ -81,6 +82,17 @@ func (precondition *ScalePrecondition) Validate(controller *api.ReplicationContr
 	return nil
 }
 
+// ValidateJob ensures that the preconditions match.  Returns nil if they are valid, an error otherwise
+func (precondition *ScalePrecondition) ValidateJob(job *experimental.Job) error {
+	if precondition.Size != -1 && *job.Spec.Parallelism != precondition.Size {
```

Comment:
An object that isn't defaulted or goes through conversions is invalid.  it shouldn't be in this code path.